sdk/serve.go

```diff
@@ -0,0 +1,71 @@
+package sdk
+
+import (
+	"github.com/cloudquery/cloudquery/cmd"
+	"github.com/cloudquery/cloudquery/plugin"
+	"github.com/hashicorp/go-hclog"
+	goplugin "github.com/hashicorp/go-plugin"
+	"log"
+	"os"
+)
+
+
+type ServeOpts struct {
+	// Required: Name of provider.
+	Name string
+
+	// Required: Provider is the actual provider that will be served.
+	Provider plugin.CQProvider
+
+	// Optional: Logger is the logger that go-plugin will use.
+	Logger hclog.Logger
+
+	// Optional: Set NoLogOutputOverride to not override the log output with an hclog
+	// adapter. This should only be used when running the plugin in
+	// acceptance tests.
+	NoLogOutputOverride bool
+}
+
+
+func ServePlugin(opts ServeOpts) {
+
+	if opts.Name == "" {
```

Comment:
if it's requried why not put it in the function signature and the options in the struct?

Reply:
It allows for better backwards compatibility